IdentityServer/Core/PersistentGrantStore.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using IdentityServer.Infrastructure.Data;
+using IdentityServer4.Models;
+using IdentityServer4.Stores;
+using Microsoft.EntityFrameworkCore;
+
+namespace IdentityServer.Core
+{
+    public class PersistentGrantStore : IPersistedGrantStore
+    {
+        private readonly PersistentGrantDbContext _context;
+
+        public PersistentGrantStore(PersistentGrantDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<PersistedGrant> GetAsync(string key)
+        {
+            var grants = await _context.PersistedGrants.AsNoTracking().FirstOrDefaultAsync(g => g.Key == key);
```

Comment:
Why didn't you just return await? Are there any advantages of this way?

Reply:
It did throw an exception a la :
* The instance of entity type 'Product' cannot be tracked because another instance with the same key value is already being tracked 

Adding AsNoTracking fixed the problem as advised in this SO thread( https://stackoverflow.com/questions/49798566/the-instance-of-entity-type-product-cannot-be-tracked-because-another-instance )